ClinicalTrial_SE_curation.Rmd

```diff
@@ -0,0 +1,320 @@
+---
+title: "Curation of Clinical trial Sumamrized Experiment objects"
+author: "Nikta Feizi, Christopher Eeles, Sisira Nair"
+date: "`r format(Sys.time(), '%a/%d/%b')`"
+output: html_document
+---
+#Please note:
+Parameters of "run_all" function are data specific and must be defined by looking at the metadata outside of the function. 
+Run "assay_data <- as.data.frame(exprs(rma.norm)) " to find "delim" which is the delimiter that separates the GSM IDs from the assay_data column names.   
+Pheno_data  column names must be used to assign column names to pre-defined columns (in "prior" vector). 
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+
+```{r}
+# configure download options for this session
+options(encoding = "UTF-8" , timeout = 1000)
+work_dir = setwd(paste0(getwd()))
+```
+
+```{r}
+# fetch Gencode v33 annotations from BHKLAB-Pachyderm/Annotations
+gencode_url <- "https://github.com/BHKLAB-Pachyderm/Annotations/raw/master/Gencode.v33.annotation.RData"
+gencode_file <- file.path(work_dir, "Ensembl.v99.annotation.RData")
+download.file(gencode_url, destfile=gencode_file)
+load("Ensembl.v99.annotation.RData")
```

Comment:
Suggestion:

TL;DR - Always use `load("path_to_file.RData", verbose=TRUE)`

Loading data from .RData files can be a bit confusing because there is no guarantee that the loaded objects have names related to the name of the file. This is why it is almost always preferable to save data to an `.rds` file, where you assign it to a variable during loading.

I obviously don't expect you to change the data object for this review, but a good work around to make sure it is clear how all objects in your script were instantiated is to set the `verbose=TRUE` flag when calling the `load` function. This will print the names of the loaded objects, making it clear where those object originated.

Reply:
The RData file - Ensembl.v99.annotation.RData is from https://github.com/BHKLAB-Pachyderm/Annotations and we might have to change it to rds. I have created an issue in that repo now.
For now, adding `verbose = TRUE`.